server/DataLouna/LounaAdmin/views.py

```diff
@@ -293,33 +451,330 @@ class getUser(APIView):
+                "site_id",
+                in_=openapi.IN_QUERY,
+                description="ID на сайте",
+                required=True,
+                type=openapi.TYPE_STRING,
+            ),
+            openapi.Parameter(
+                name="status",
+                required=True,
+                type=openapi.TYPE_STRING,
+                in_=openapi.IN_QUERY,
+                description="ZERO(Нет аккаунта на сайте)"
+                            "FIRST(Не оформил триал),"
+                            "SECOND(Триал оформлен),"
+                            "THIRD(Оформил (продлил?) подписку)"
+                            "FOURTH(Карта удалена сразу)",
+            ),
+            openapi.Parameter(
+                name="subscription_end_date",
+                required=False,
+                type=openapi.TYPE_STRING,
+                in_=openapi.IN_QUERY,
+                description="YYYY-MM-DD",
+            ),
+            openapi.Parameter(
+                name="Authorization",
+                description="Authorization token",
+                required=True,
+                type=openapi.TYPE_STRING,
+                in_=openapi.IN_HEADER,
+            ),
+        ],
+        responses=response_schema_dict,
+    )
+    def get(self, request):
+        site_id = self.request.query_params.get("site_id")
+        status_ = self.request.query_params.get("status")
+        subscription_end_date = self.request.query_params.get("subscription_end_date")
+
+        status_check = modelUser.objects.filter(site_id=site_id)
+
+        if status_check.exists() and status_ in ("ZERO", "FIRST", "SECOND", "THIRD", "FOURTH"):
+            status_check.update(
+                site_id=site_id,
+                subscription_status=status_,
+            )
+            if status_ != "ZERO" and status_ != "FIRST" and subscription_end_date:
+                status_check.update(
+                    site_id=site_id,
+                    subscription_status=status_,
+                    subscription_end_date=subscription_end_date
+                )
+                return Response("Запись произведина успешно", status=status.HTTP_200_OK)
+            elif status_ != "ZERO" and status_ != "FIRST" and subscription_end_date is None:
+                return Response("Если статус не равен ZERO, то нужно передавать subscription_end_date",
+                                status=status.HTTP_400_BAD_REQUEST)
+            else:
+                status_check.update(
+                    site_id=site_id,
+                    subscription_status=status_,
+                    subscription_end_date=None
+                )
+                return Response(f"Статус равен {status_}")
+        elif status_ not in ("ZERO", "FIRST", "SECOND", "THIRD", "FOURTH"):
+            return Response("Неверный статус", status=status.HTTP_400_BAD_REQUEST)
+        else:
+            return Response("Пользователь не найден", status=status.HTTP_404_NOT_FOUND)
```

Comment:
```suggestion
            return Response("User not found", status=status.HTTP_404_NOT_FOUND)
```
а ещё лучше вернуть json как было сделано выше (619)